### ISSUE TITLE
Template tests fail to parse when referencing a subproperty without parenthesis

### ISSUE THREAD

**Issue by micoomsft** [pallets/jinja#930]
## Expected Behavior
Sub properties of properties in the template data should be able to be referenced as the additional test parameter of a test without using parenthesis 

## Actual Behavior
The test fails to parse if a sub property of a property (with a dictionary value) in the template data is referenced as the additional argument of a test

## Template Code
This template fails parsing:

```jinja
{% for target in Iterator if target.property is eq Value.Subproperty %}
    {{target.property}}
{% endfor %}
```
This template succeeds:

```jinja
{% for target in Iterator if target.property is eq Value %}
    {{target.property}}
{% endfor %}
```
This template also succeeds:

```jinja
{% for target in Iterator if target.property is eq(Value.Subproperty) %}
    {{target.property}}
{% endfor %}
```

## Full Traceback
```pytb
File "c:\Users\Mike\Documents\projects\jinja2visitor\notes\test.py", line 80, in <module>
    print env.parse(template)
  File "C:\python27\lib\site-packages\jinja2\environment.py", line 493, in parse
    self.handle_exception(exc_info, source_hint=source)
  File "C:\python27\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "<unknown>", line 1, in template
jinja2.exceptions.TemplateSyntaxError: expected token 'end of statement block', got '.'
```

## Your Environment
* Python version: 2.7.11
* Jinja version: 2.10

**Comment by bkiu** [comment]
I'll work on this at PyCon 2019 sprints.

**Comment by eplata31** [comment]
Hi, I've been doing some research on this and I'm wondering if I'm on the right track.
I founded the 'parse_if()' has inside the 'parse_tuple()' and that function specifies something about 'explicit_parenthesis' and when it doesn't have the parenthesis the return value is odd to me, maybe that is related?

**Comment by eplata31** [comment]
Hello! Asking for a review :D

**Comment by jbaehr** [comment]
@kevin-brown could you please consider this fix for a potential v2.10.2 release, too? As far as I see this has only been merged to master... Are there any planes for an additional 2.10 bugfix release "soon"?
